tasks/ThrusterLimitationTask.cpp

```diff
@@ -27,41 +27,53 @@ bool ThrusterLimitationTask::configureHook()
         return false;
     }
     m_limits = _limits.get();
-    if (m_limits.elements.empty()) {
-        // Initialize speed limits as infinity
-        JointLimits infinity;
-        JointLimitRange range;
-        range.Speed(-base::infinity<float>(), base::infinity<float>());
-        infinity.elements.push_back(range);
-        m_limits = infinity;
-    }
-
     return true;
 }
 
-bool ThrusterLimitationTask::startHook()
+bool validateCommandType(JointState const& state, JointLimitRange const& limit)
 {
-    if (!ThrusterLimitationTaskBase::startHook()) {
+    if(state.hasPosition() && (!limit.min.hasPosition() || !limit.max.hasPosition()))
+    {
+        return false;
+    }
+    if(state.hasSpeed() && (!limit.min.hasSpeed() || !limit.max.hasSpeed()))
+    {
+        return false;
+    }
+    if(state.hasRaw() && (!limit.min.hasRaw() || !limit.max.hasRaw()))
+    {
+        return false;
+    }
+    if(state.hasEffort() && (!limit.min.hasEffort() || !limit.max.hasEffort()))
+    {
+        return false;
+    }
+    if(state.hasAcceleration() && (!limit.min.hasAcceleration() || !limit.max.hasAcceleration()))
+    {
```

Comment:
This forces us to always set a saturation limit, which can probably bite us back somehow